docs/get-started.md

```diff
@@ -116,7 +105,145 @@ Execution succeeded (Time spent: 154/793 us)
 
 ---
 
-### Step 3 Simple Query Demo
+### From Source(Linux)
+
+**Prerequisite Tools**
+
+Nebula Graph is written in C++14, so it requires a complier supporting C++14 features.
+
+3rd-party Libraries
+
+-	autoconf
+-	automake
+-	libtool
+-	cmake
+-	bison
+-	unzip
+-	boost
+-	gperf
+-	krb5
+-	openssl
+-	libunwind
+-	ncurses
+-	readline
+
+It is recommended to install g++ 5 or higher Linux system, such as Fedora 29.
+Currently, we are using `git-lfs` to store the 3rd-party libraries so make sure
+`git-lfs` have been installed before fetching the source code.
+
+**Fetch from GitHub**
+
+```
+git clone https://github.com/vesoft-inc/nebula.git
+```
+
+**Compiling**
+
+```
+cmake ./
+```
+
+The default installation is in /usr/local path. To specify the installation path,
+use:
+
+```
+cmake -DCMAKE_INSTALL_PREFIX=$your_nebula_install_dir
+```
+
+to replace the `$your_nebula_install_dir` here
+
+Then run the following command:
+
+```
+make && make install
+```
+
+**Running**
+
+Configure nebula-metad.conf
+
+In your Nebula installation directory, run
+
+```
+cp etc/nebula-metad.conf.default etc/nebula-metad.conf
+```
+
+Modify configurations in nebula-metad.conf, such as
+
+- local_ip
+- port
+- ws_http_port metaservice HTTP
+- ws_h2_port metaservice HTTP2
+
+
+Configure nebula-storaged.conf
+
+```
+cp etc/nebula-storaged.conf.default etc/nebula-storaged.conf
+```
+
+Modify configurations in nebula-storaged.conf, such as
+
+- local_ip
+- port
+- ws_http_port metaservice HTTP
+- ws_h2_port metaservice HTTP2
+
+Configure nebula-graphd.conf
+
+```
+cp etc/nebula-graphd.conf.default etc/nebula-graphd.conf
+```
+
+Modify configurations in nebula-storaged.conf, such as
+
+- local_ip
+- port
+- ws_http_port metaservice HTTP
+- ws_h2_port metaservice HTTP2
+
+**Start service**
+
+```
+script/nebula.service start all
+```
+
+Make sure all the services are working
+
+```
+script/nebula.service status all
+```
+
+**Connect to Nebula**
+
+```
+bin/nebula -u=user -p=password
+```
+
+- -u is user name, default user
+- -p is password, default password
+
+Add host after connection:
+
+```
+bin/nebula -u=user -p=password
+> ADD HOSTS $storage_ip:$storage_port
+```
+
+Replace the `$storage_ip` and `$storage_port` here according to the `local_ip`
+and `port` in nebula-storaged.conf
+
+Then you’re now ready to start using Nebula Graph.
+
+---
+
+## Update Nebula Graph
+
+Upgrading Nebula Graph is as easy as downloading and replacing the executable you’ve placed in your PATH, if you are using Docker, run docker upgrade nebula.
```

Comment:
@jude-zhu stop all the services (try the command).
re-install by binary, ...
For docker, re-pull the lastest images:
docker pull vesoft/nebula-graph:latest